Makes parent context dependency for GroupBuilder class injectible

MMSs opened the pull request:
This update addresses https://github.com/StoutLogic/acf-builder/issues/96.

Suggestions for a different implementation are welcome.

stevep commented:
Hey @MMSs thanks for tackling this. I'm not opposed. It will definitely make FieldsBuilder more extendable for more user land customization. 

One question I have is what do you think about making it the 4th argument instead, and making it optional, so that it wouldn't drastically change the function signature and wouldn't break backward compatibility.

So like GroupBuilder would be like:
```php
/**
 * @param string $name Field name
 * @param string $type Field name
 * @param array $config Field configuration
 * @param FieldsBuilder $builder
 */
public function __construct($name, $type = 'group', $config = [],  FieldsBuilder $builder = null)
{
    parent::__construct($name, $type, $config);
    $this->fieldsBuilder = $builder ?: (new FieldsBuilder);
    $this->fieldsBuilder->setParentContext($this);
}
``` 

and then change the `addGroup` function to be:
```php
public function addGroup($name, array $args = [])
{
    return $this->initializeField(new GroupBuilder($name, 'group', $args,  new self($name)));
}
```

MMSs commented:
Hey @stevep,

I appreciate your response to my suggestion, and I might have another later :)

I agree with you on not introducing a breaking change in constructor's signature, and I was going for an implementation exactly like your suggestion, but I was hesitant as it will not remove the instantiation from the constructor which is the original suggestion, but I don't mind what you suggested either, so I changed the implementation now to match your suggestion.

I also added tests to cover the conditional dependancy injection.

MMSs commented:
I'm closing this PR, as it appears there's no interest in merging it.
